### ISSUE TITLE
Upload via public folder fails

### ISSUE THREAD

**Issue by Peac** [nextcloud/server#4604]
### Steps to reproduce
1. Create a new folder.
2. Share folder via Link with "Allow upload and editing" enabled.
3. Log off
4. Go to the link and upload a file
5. Download the uploaded file via webinterface

### Expected behaviour
The file should be downloadable/accessible.

### Actual behaviour
The file is not downloadable/accessible. And a wrong filesize is shown.

### Server configuration

**Operating system**:
Ubuntu 16.04.02
**Web server:**
apache  2.4.18
**Database:**
mysql 5.7.18
**PHP version:**
7.0.15
**Nextcloud version:** (see Nextcloud admin page)
11.0.3
**Updated from an older Nextcloud/ownCloud or fresh install:**
updated
**Where did you install Nextcloud from:**
Downloaded from nextcloud.com
**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
  - calendar: 1.5.2
  - comments: 1.1.0
  - contacts: 1.5.3
  - dav: 1.1.1
  - encryption: 1.4.1
  - federatedfilesharing: 1.1.1
  - files: 1.6.1
  - files_pdfviewer: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - files_videoplayer: 1.0.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.0
  - notifications: 1.0.1
  - password_policy: 1.1.0
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - updatenotification: 1.1.1
  - workflowengine: 1.1.1

```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***.***"
        ],
        "datadirectory": "\/var\/www\/nextcloud\/data",
        "overwrite.cli.url": "https:\/\/***.***",
        "dbtype": "mysql",
        "version": "11.0.3.2",
        "dbname": "***",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Berlin",
        "installed": true,
        "appcodechecker": false,
        "updatechecker": true,
        "check_for_working_webdav": true,
        "check_for_working_wellknown_setup": true,
        "check_for_working_htaccess": true,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\SVG",
            "OC\\Preview\\TIFF",
            "OC\\Preview\\Font",
            "OC\\Preview\\Movie"
        ],
        "cipher": "AES-256-CFB",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "maintenance": false,
        "versions_retention_obligation": "auto",
        "theme": "",
        "loglevel": 2,
        "updater.release.channel": "stable"
    }
}

```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
yes
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Firefox 53.0 (32bit)
**Operating system:**
Windows 7 / Ubuntu 16.04/14.04
#### Web server error log
<details>
<summary>Web server error log</summary>

```
none
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
![unbenannt](https://cloud.githubusercontent.com/assets/6718360/25558070/80bb391e-2d1f-11e7-925b-d546686e1c9f.PNG)

```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
none
</details>

**Comment by Peac** [comment]
Here is another one with this issue:
https://help.nextcloud.com/t/photos-uploaded-via-public-upload-not-accessible-with-encryption-module/11962

**Comment by MorrisJobke** [comment]
Looks similar to what https://github.com/nextcloud/server/pull/4277 fixed. 

@Peac Are the files uploaded before or after the upgrade to 11.0.3 (we backported the fix: https://github.com/nextcloud/server/pull/4278)

@icewind1991 Are more steps needed to fix those permissions? Maybe a files scan?

**Comment by Peac** [comment]
I uploaded the files after the upgrade. With 11.0.2 it worked just fine for me.

**Comment by Peac** [comment]
Just tested it on a demo instance. Without encryption it works flawless, but with encryption enabled it shows the wrong file size (8 KB instead of 1 KB) and the download of the file fails.

**Comment by MorrisJobke** [comment]
cc @schiessle

**Comment by GetBoz** [comment]
Same error on my server and also on demo.nextcloud.com !!!

### Steps to reproduce
1. Enable "Default encryption module" app
2. Activate "Enable server-side encryption" in admin settings
3. Log-out and log-in again
4. Create a folder
5. Share this folder with "Share link" 
6. Add "Allow upload and editing"
7. Open link in another browser (NOT with the browser you are logged-in)
8. Upload a file from this browser 

### Expected behaviour
The uploaded file should be readable

### Actual behaviour
The uploaded file is not readable (from both sides)

**Comment by GetBoz** [comment]
Fixed in 12.0.0.
But i hope it will be fixed in 11.0.4, because encryption and share by link are core features of Nextcloud.
